tests/spark/test_spark_connect_model_export.py

```diff
@@ -0,0 +1,146 @@
+    return new_df.toPandas()["prediction"]
+
+
+@pytest.fixture(scope="module")
+def spark():
+    spark = _get_spark_connect_session()
+    yield spark
+    spark.stop()
+
+
+@pytest.fixture(scope="module")
+def iris_df(spark):
+    X, y = datasets.load_iris(return_X_y=True)
+    spark_df = spark.createDataFrame(zip(X, y), schema="features: array<double>, label: long")
+    return spark_df.toPandas(), spark_df
+
+
+@pytest.fixture(scope="module")
+def spark_model(iris_df):
+    iris_pandas_df, iris_spark_df = iris_df
+    scaler = StandardScaler(inputCol="features", outputCol="scaled_features")
+    lr = LogisticRegression(maxIter=50, numTrainWorkers=2, learningRate=0.001)
+    pipeline = Pipeline(stages=[scaler, lr])
+    # Fit the model
+    model = pipeline.fit(iris_spark_df)
+    preds_pandas_df = model.transform(iris_pandas_df.copy(deep=False))
+    return SparkModelWithData(
+        model=model,
+        spark_df=iris_spark_df,
+        pandas_df=iris_pandas_df,
+        predictions=preds_pandas_df,
+    )
+
+
+@pytest.fixture
+def model_path(tmp_path):
+    return os.path.join(tmp_path, "model")
+
+
+def test_model_export(spark_model, model_path):
+    mlflow.spark.save_model(spark_model.model, path=model_path)
+    # 1. score and compare reloaded sparkml model
+    reloaded_model = mlflow.spark.load_model(model_uri=model_path)
+    preds_df = reloaded_model.transform(spark_model.pandas_df.copy(deep=False))
+    pd.testing.assert_frame_equal(spark_model.predictions, preds_df, check_dtype=False)
+
+    m = pyfunc.load_model(model_path)
+    # 2. score and compare reloaded pyfunc
+    preds2 = m.predict(spark_model.pandas_df.copy(deep=False))
+    pd.testing.assert_series_equal(spark_model.predictions["prediction"], preds2, check_dtype=False)
+
+    # 3. score and compare reloaded pyfunc Spark udf
+    preds3 = score_model_as_udf(
+        model_uri=model_path, pandas_df=spark_model.pandas_df, result_type=LongType()
+    )
+    pd.testing.assert_series_equal(spark_model.predictions["prediction"], preds3, check_dtype=False)
+
+
+def test_sparkml_model_log(spark_model):
+    with mlflow.start_run():
+        model_info = mlflow.spark.log_model(
+            artifact_path="model",
+            spark_model=spark_model.model,
+        )
+    model_uri = model_info.model_uri
+
+    reloaded_model = mlflow.spark.load_model(model_uri=model_uri)
+    preds_df = reloaded_model.transform(spark_model.pandas_df.copy(deep=False))
+    pd.testing.assert_frame_equal(spark_model.predictions, preds_df, check_dtype=False)
+
+
+def test_pyfunc_serve_and_score(spark_model):
+    artifact_path = "model"
+    with mlflow.start_run():
+        mlflow.spark.log_model(spark_model.model, artifact_path)
+        model_uri = mlflow.get_artifact_uri(artifact_path)
+
+    input_data = pd.DataFrame(
+        {"features": spark_model.pandas_df.features.map(lambda x: x.tolist())}
```

Comment:
```suggestion
        {"features": spark_model.pandas_df["features"].map(list)}
```

Does this work?

Reply:
btw why do we need this line?

Reply:
Because `pyfunc_serve_and_score_model` does not support input pandas dataframe containing column of "numpy array" type. Converting the column values to "list" type works.